src/platform/packages/private/kbn-repo-packages/modern/types.ts

```diff
@@ -181,6 +181,10 @@ export interface PluginSelector {
    * When set to true, only select plugins which have browser-side components
    */
   browser?: boolean;
+  /**
+   * When defined, only select plugins that belong to the specified groups
+   */
+  allowlistPluginGroups?: readonly string[];
```

Comment:
should we not further constrain the type declarations of `allowlistPluginGroups` wherever it's used to the 4 solution types?

Reply:
at some point, `common` is enabled besied the 4 solution types. I hope I'm not mixing these up

Reply:
Yeah, but I decided not to add it on that list. The idea is that you can not explicitly enable 'common'.
The same way you cannot explicitly set a `"group": "common"` in the manifest.

At the moment, 'common' is only used for test and example plugins, which don't have a group. We can enable / disable those based on the flags and prod vs not.
I'll add a check to ensure that all the other plugins have a group.